homeassistant/components/media_player/sonos.py

```diff
@@ -762,6 +768,11 @@ def is_volume_muted(self):
         """Return true if volume is muted."""
         return self._player_volume_muted
 
+    @property
+    def shuffle(self):
+        """Shuffling state."""
+        return True if self._player.play_mode == 'SHUFFLE' else False
```

Comment:
This seems to do I/O and thus is not allowed in a property. Instead, the value should be cached inside `update` and the cached value be returned here.